src/parser.js

```diff
@@ -2149,6 +2151,9 @@ export class GenericParser extends Tokenizer {
           if (this.allowYieldExpression && token.value === 'yield') {
             throw this.createError(ErrorMessages.ILLEGAL_YIELD_IDENTIFIER);
           }
+          if (this.allowAwaitExpression && token.value === 'await') {
+            throw this.createError(ErrorMessages.ILLEGAL_AWAIT_IDENTIFIER);
```

Comment:
I think you probably need this in the previous (`if (this.eat(TokenType.ASSIGN)) {`) branch too, to prevent

```js
async function f () { return {await = 0} = {} }
```
from parsing. Either way, can you add that as a test?

Reply:
Added